api/app/src/routes/user.ts

```diff
@@ -129,28 +156,50 @@ router.delete(
     const cxId = getCxIdOrFail(req);
     const connectedUser = await getConnectedUserOrFail({ id: userId, cxId });
 
-    const providerOAuth2 = providerOAuth2OptionsSchema.safeParse(req.query.provider);
+    if (await revokeUserProviderAccess(connectedUser, req.query.provider)) {
+      return res.sendStatus(status.OK);
+    }
+  })
+);
 
-    // TODO #249: implement garmin revoke support
-    // const providerOAuth1 = providerOAuth1OptionsSchema.safeParse(
-    //   req.query.provider
-    // );
+/**
+ * DELETE /user/delete
+ *
+ * Revoke access to all providers and permanently delete the user
+ *
+ * @param {string} req.query.userId      The internal user ID.
+ *
+ * @return  {{success: boolean}}      If successfully removed.
+ */
+router.delete(
+  "/delete",
+  asyncHandler(async (req: Request, res: Response) => {
+    const userId = getUserIdFrom("query", req).orFail();
+    const cxId = getCxIdOrFail(req);
+    const connectedUser = await getConnectedUserOrFail({ id: userId, cxId });
 
-    if (providerOAuth2.success) {
-      await Constants.PROVIDER_OAUTH2_MAP[providerOAuth2.data].revokeProviderAccess(connectedUser);
+    if (connectedUser.providerMap) {
+      const connectedProviders = Object.keys(connectedUser.providerMap).map(key => {
+        return key;
+      });
+      try {
+        await Promise.all(
+          connectedProviders.map(async provider => {
+            await revokeUserProviderAccess(connectedUser, provider);
+          })
+        );
+      } catch (err) {
+        throw new Error(`Unable to revoke access to all providers: ${err}`);
```

Comment:
I recommend we keep the text of the error w/o the result of `${err}` so its cleaner to read on Sentry.

Also, I'd:
- use `capture.error(err, { extra: { ... } });` to send the error to Sentry from here instead of relying on the default error handler; reason being here we have more info to share on `extra`, like `userId` and number of failed/succeeded revokes; _and_
- log the error with the context information we have here; _and_
- return 500 from here so we don't get two errors on Sentry/Slack